Remove old and deprecated UIProcess functions

LumioseSil opened the pull request:
#### 8ae8a3d07be227426026daf2d5f1a113acd3d0c4
<pre>
Remove old and deprecated UIProcess functions
<a href="https://bugs.webkit.org/show_bug.cgi?id=251879">https://bugs.webkit.org/show_bug.cgi?id=251879</a>

Reviewed by NOBODY (OOPS!).

The code in question is literally code that was commented to be removed
under certain conditions that have been met years ago, such as no longer
compiling for Safari 8 and when Mavericks support is gone.

This is very very old code that is the removal of which is obvious but
has yet to be done.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ae8a3d07be227426026daf2d5f1a113acd3d0c4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109247 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/18332 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/42058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/87/builds/773 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/118477 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/113136 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/19789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/9631 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101491 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115002 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/19789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/42058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/43011 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/19789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/42058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/84753 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11105 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/42058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/11830 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/9631 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/17196 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/42058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13454 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on LayoutTests/imported/w3c/web-platform-tests/service-workers/service-worker/resources/testharness-helpers.js:
```diff
@@ -96,28 +96,6 @@ function assert_array_objects_equals(actual, expected, description) {
-//
-// FIXME: Remove assert_will_be_idl_attribute once we complete the transition
-// of moving the DOM attributes to prototype chains.  (http://crbug.com/43394)
-function assert_will_be_idl_attribute(object, attribute_name, description) {
```
A PR needs to go upstream for this. Modifying it downstream won't be useful otherwise since it will get overwritten during the next resin.

cdumez commented on Source/WebKit/UIProcess/API/C/WKContext.cpp:
```diff
@@ -457,16 +457,6 @@ void WKContextAllowSpecificHTTPSCertificateForHost(WKContextRef, WKCertificateIn
-    WebKit::toImpl(contextRef)->disableProcessTermination();
-}
-
-void WKContextEnableProcessTermination(WKContextRef contextRef)
```
How did you confirm that all these changes to our C API were OK?
Do we know for sure there are no longer any (Apple-internal) projects relying on these?

AZero13 replied:
I am not sure to be honest.

hortont424 reviewed (changes requested):
This patch changes *way* too many unrelated things; some of the changes are good (and some I can't tell), but should be many separate PRs.

LumioseSil committed:
Remove old and deprecated UIProcess functions
https://bugs.webkit.org/show_bug.cgi?id=251879

Reviewed by NOBODY (OOPS!).

The code in question is literally code that was commented to be removed
under certain conditions that have been met years ago, such as no longer
compiling for Safari 8 and when Mavericks support is gone.

This is very very old code that is the removal of which is obvious but
has yet to be done.
